gibberish_server/controllers/room_controller.js

```diff
@@ -24,23 +23,32 @@ class RoomController {
 
     static joinRoom(req, res) {
         // TODO: race conditionSTATE.GAME_WAITING
-        let r = roomModel.Room.deserializeRoom(roomModel.getRoomInfoAsJson(roomId));
-        if (r.state != STATE.GAME_WAITING) return "";
-        for (let p in r.players) {
-            if (p.name == nickname) {
-                return "";
-            }
-        }
-        r.players.push(new Player(nickname))
-        roomModel.saveRoom(r, client);
-        res.sen(r.players[r.length - 1].name);
+        const nickname = req.body.nickname
+        const roomId = req.body.roomId
+        roomModel.getRoomInfoAsJson(roomId)
```

Comment:
i think its better to use getRoomInfoAsObject, and refactor getRoomInfoAsObject to use promise instead